.github/workflows/publish-doc.yaml

```diff
@@ -26,7 +26,7 @@ jobs:
           pip install mkdocs-material mike
       - name: Build
         run: |
-          mike deploy 1.0.5 latest --update-aliases --push
+          mike deploy v1.0.6-rc1 latest --update-aliases --push
```

Comment:
These changes seems not related to TLS passthrough feature.  Can you separate them into different CR? thanks

Reply:
This one is a little bit special, need to bump up the doc version here. otherwise the the doc version 1.0.5 will include the tls-passthrough.md, but the the controller v1.0.5 actually don't support the TLSRoute.

https://www.gateway-api-controller.eks.aws.dev/1.0.5/


What I am think is we can improve the whole Github Action (CI pipeline) publish-doc.yaml (in the future),  for example, main branch code change trigger the `mike deploy dev latest`, and branch `release-xxxxx` trigger the `mike deploy v1.x.x `